protocol/bc/transaction.go

```diff
@@ -267,6 +250,14 @@ func (tx *TxData) HashForSig(idx uint32) Hash {
 	return NewSigHasher(tx).Hash(idx)
 }
 
+func (tx *Tx) OutputID(outputIndex int) OutputID {
+	return ComputeOutputID(tx.Hash, uint32(outputIndex))
```

Comment:
Good call. I moved the typecase to callsites where the index is obviously bounded as it's coming from a loop over a reasonably-sized data structure. Fixed here: 8c3bab7